homeassistant/components/matrix/__init__.py

```diff
@@ -219,7 +219,9 @@ async def handle_startup(event: HassEvent) -> None:
                 loop_sleep_time=1_000,
             )  # milliseconds.
 
-        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, handle_startup)
+        self.hass.bus.async_listen_once(
```

Comment:
There is a race in the test where the test expected it won't send anything in time, with run_immediately it can get it sent in time